src/ucp/tag/rndv.c

```diff
@@ -181,7 +187,10 @@ ucs_status_t ucp_tag_send_start_rndv(ucp_request_t *sreq)
         return status;
     }
 
-    if (ucp_ep_is_tag_offload_enabled(ucp_ep_config(ep))) {
+    md_attr = ucp_ep_md_attr(ep, sreq->send.lane);
+
+    if (ucp_ep_is_tag_offload_enabled(ucp_ep_config(ep)) &&
+        md_attr->cap.reg_mem_types & UCS_BIT(sreq->send.mem_type)) {
```

Comment:
please move this check to `ucp_tag_offload_start_rndv`. When tag offload enabled we can't use regular protocols (otherwise messages can be matched in a wrong order by peer). Instead need to select `ucp_tag_offload_sw_rndv` inside `ucp_ep_is_tag_offload_enabled`

Reply:
seems you already moved this in https://github.com/openucx/ucx/pull/4499 .  had fix assert in ucp_tag_offload_start_rndv.